app/controllers/api/users_controller.rb

```diff
@@ -3,7 +3,13 @@ class Api::UsersController < Api::BaseController
   before_action :authorize_resource!
 
   def destroy
-    User.find(params.require(:id)).soft_delete!
```

Comment:
then the solution should not be a 1-off fix ... if api 404 renders as html then we need a general fix ...